src/Microsoft.ML/Runtime/EntryPoints/TrainTestMacro.cs

```diff
@@ -117,10 +129,13 @@ public sealed class Output
             subGraphRunContext.RemoveVariable(dataVariable);
 
             // Change the subgraph to use the model variable as output.
-            varName = input.Outputs.Model.VarName;
+            varName = input.Outputs.UseTransformModel ? input.Outputs.TransformModel.VarName : input.Outputs.Model.VarName;
             if (!subGraphRunContext.TryGetVariable(varName, out dataVariable))
                 throw env.Except($"Invalid variable name '{varName}'.");
-            string outputVarName = node.GetOutputVariableName(nameof(Output.PredictorModel));
+
+            string outputVarName = input.Outputs.UseTransformModel ? node.GetOutputVariableName(nameof(Output.TransformModel)) : 
```

Comment:
>node.GetOutputVariableName [](start = 69, length = 26)

you can probably move it forward in your statement and put ternary operator inside it.
node.GetOuputVariableName( input.Outputs.UseTransformModel? nameof(Output.TransformModel): nameof(Output.PredictorModel));

Reply:
this is just fine.

---
In reply to: [191029906](https://github.com/dotnet/machinelearning/pull/212#discussion_r191029906) [](ancestors = 191029906)